### PR TITLE
secure_storage: Fix counter bug

### PR DESCRIPTION
**Pull request by Vge0rge**
The counter used as a nonce for the secure storage was 16 bytes but the implementation on the AEAD backend expected a 12 byte nonce. This corrupted the key material when the nonce was written.
This fixes the issue by making the counter to be 12 bytes as expected by the AEAD backend. It also ensures the AEAD backend and the counter have the same length and fails at runtime without corrupting any buffers if they differ.

### CODE REVIEW

File: subsys/secure_storage/src/aead/aead_ctr_nonce.c [hunk 1]
```diff
@@ -35,7 +32,7 @@ psa_status_t secure_storage_get_nonce(uint8_t *nonce, size_t nonce_len)
 		return PSA_ERROR_INVALID_ARGUMENT;
 	}
 
-	if (nonce_len > NONCE_MAX_LENGTH) {
+	if (nonce_len != sizeof(g_nonce)) {
 		return PSA_ERROR_NOT_SUPPORTED;
 	}
```
Comment (vili-nordic) [inline subsys/secure_storage/src/aead/aead_ctr_nonce.c:38]: comment on line 43 is wrong

Comment (Vge0rge) [reply subsys/secure_storage/src/aead/aead_ctr_nonce.c:38]: true! fixed!

File: subsys/secure_storage/src/aead/aead_ctr_nonce.c [hunk 2]
```diff
@@ -35,24 +32,23 @@ psa_status_t secure_storage_get_nonce(uint8_t *nonce, size_t nonce_len)
...
 
 	if (nonce_len == 0) {
 		return PSA_SUCCESS;
 	}
 
-	/* Incrementing is implemented by using 2 64bit numbers */
+	/* Incrementing is implemented by using a 64bit an a 32bit number */
```
Comment (frkv) [inline subsys/secure_storage/src/aead/aead_ctr_nonce.c:43]:
```suggestion
	/* Incrementing is implemented by using a 64-bit and a 32-bit number */
```

**Review by maciejbaczmanski — APPROVED** [review]
Tested on OT CLI samples on nrf52 and nrf53, Issue with asserts after board restart does not reproduce anymore. Thanks!

**Review by ArekBalysNordic — APPROVED** [review]
It repairs the issue with resetting nRF53/52 DKs also in Matter samples when PSA crypto is enabled.

**Review by vili-nordic — APPROVED** [review]

**Pull request merged**